defs/scenarios/storage/ceph/ceph-osd/PG_overdose.yaml

```diff
@@ -0,0 +1,25 @@
+checks:
+  health_warning:
+    property:
+      path: hotsos.core.plugins.storage.ceph.CephCluster.health_status
+      ops: [[ne, HEALTH_OK]]
+  # this one will run on OSD sos
+  ceph_osd_withhold_creation:
+    input:
+      path: [var/log/ceph/ceph*osd*.log, var/log/ceph/ceph.log]
+    search:
+      expr: '.+ (osd.+) .+ maybe_wait_for_max_pg withhold creation of pg .+: (.+) >= (.+)'
+
+conclusions:
+  pending_creating_pgs:
+    decision:
+      - health_warning
+      - ceph_osd_withhold_creation
+    raises:
+      type: CephOSDWarning
+      message: >-
+        ceph is trying to create {} PGs on {} but the hard limit is {}.
```

Comment:
Since we not have a way yet to apply logic to a value extracted from a file, can we at least say in the message that if the number of pending PGs is above a certain point it requires more investigation before applying the workaround.

Reply:
Done. I put it in the osd message because we print out the limit in that one.